Source/Scene/MetadataTableProperty.js

```diff
@@ -0,0 +1,610 @@
+      return ComponentDatatype.UNSIGNED_SHORT;
+    case MetadataType.INT32:
+      return ComponentDatatype.INT;
+    case MetadataType.UINT32:
+      return ComponentDatatype.UNSIGNED_INT;
+    case MetadataType.FLOAT32:
+      return ComponentDatatype.FLOAT;
+    case MetadataType.FLOAT64:
+      return ComponentDatatype.DOUBLE;
+  }
+}
+
+function requiresUnpackForGet(property) {
+  if (defined(property._unpackedValues)) {
+    return false;
+  }
+
+  var valueType = property._classProperty.valueType;
+
+  if (valueType === MetadataType.STRING) {
+    // Unpack since UTF-8 decoding is expensive
+    return true;
+  }
+
+  if (
+    valueType === MetadataType.INT64 &&
+    !FeatureDetection.supportsBigInt64Array()
+  ) {
+    // Unpack since the fallback INT64 getters are expensive
+    return true;
+  }
+
+  if (
+    valueType === MetadataType.UINT64 &&
+    !FeatureDetection.supportsBigUint64Array()
+  ) {
+    // Unpack since the fallback UINT64 getters are expensive
+    return true;
+  }
+
+  return false;
+}
+
+function requiresUnpackForSet(property, index, value) {
+  if (requiresUnpackForGet(property)) {
+    return true;
+  }
+
+  var arrayOffsets = property._arrayOffsets;
+  if (defined(arrayOffsets)) {
+    // Unpacking is required if a variable-size array changes length since it
+    // would be expensive to repack the binary data
+    var oldLength = arrayOffsets.get(index + 1) - arrayOffsets.get(index);
+    var newLength = value.length;
+    if (oldLength !== newLength) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function unpackProperty(property) {
+  property._unpackedValues = unpackValues(property);
+
+  // Free memory
+  property._arrayOffsets = undefined;
+  property._stringOffsets = undefined;
+  property._values = undefined;
+}
+
+function unpackValues(property) {
+  var i;
+  var count = property._count;
+  var unpackedValues = new Array(count);
+
+  var classProperty = property._classProperty;
+  if (classProperty.type !== MetadataType.ARRAY) {
+    for (i = 0; i < count; ++i) {
+      unpackedValues[i] = property._getValue(i);
+    }
+    return unpackedValues;
+  }
+
+  var j;
+  var offset;
+  var arrayValues;
+
+  var componentCount = classProperty.componentCount;
+  if (defined(componentCount)) {
+    for (i = 0; i < count; ++i) {
+      arrayValues = new Array(componentCount);
+      unpackedValues[i] = arrayValues;
+      offset = i * componentCount;
+      for (j = 0; j < componentCount; ++j) {
+        arrayValues[j] = property._getValue(offset + j);
+      }
+    }
+    return unpackedValues;
+  }
+
+  for (i = 0; i < count; ++i) {
+    offset = property._arrayOffsets.get(i);
+    var length = property._arrayOffsets.get(i + 1) - offset;
+    arrayValues = new Array(length);
+    unpackedValues[i] = arrayValues;
+    for (j = 0; j < length; ++j) {
+      arrayValues[j] = property._getValue(offset + j);
+    }
+  }
+
+  return unpackedValues;
+}
+
+function BufferView(bufferView, type, length) {
```

Comment:
Long term, would there be any usefulness of unifying these buffer views with the bitstreams that are used in implicit tiling?

Reply:
Good idea, I'll factor it out into something more generic

Reply:
Actually `BufferView` may be too low level to expose. However `MetadataTableProperty` can be used fairly cleanly to read availability if availability is treated as a `BOOLEAN` class property.